clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerClientInitialResponse.java

```diff
@@ -78,7 +81,7 @@ public OAuthBearerClientInitialResponse(String tokenValue, SaslExtensions extens
     public OAuthBearerClientInitialResponse(String tokenValue, String authorizationId, SaslExtensions extensions) throws SaslException {
         this.tokenValue = tokenValue;
         this.authorizationId = authorizationId == null ? "" : authorizationId;
-        this.saslExtensions = validateExtensions(extensions);
+        this.saslExtensions = extensions != null ? validateExtensions(extensions) : NO_SASL_EXTENSIONS;
```

Comment:
It may be good to add this check in `validateExtensions(extensions)` and return respective instance because it will avoid having null check for extensions in any code which will be added in future. 

```java
public static SaslExtensions validateExtensions(SaslExtensions extensions) throws SaslException {
  if(extensions == null) {
    return NO_SASL_EXTENSIONS; // nothing to validate further
  }
  // existing logic
  ....
}
```

validateExtensions(extensions) may be renamed to be explicit about returning resultant  extensions. 

Assigning `validateExtensions(extensions)` result is missed [here](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredLoginCallbackHandler.java#L234) which is not added as part of this PR but it may be addressed in this PR if possible.

Reply:
👍